src/set.js

```diff
@@ -0,0 +1,220 @@
+function Set() {
+  this.dataStore = []
+  this.size = 0
+}
+Set.prototype.add = function(element){
+  this.dataStore[this.size++] = element
+  console.log('+1')
+}
+
+Set.prototype.contains = function(){
+  return this.size > 0 ? 'true' : 'false'
+}
+
+Set.prototype.isEmpty = function(){
+  return this.size = 0 ? 'false' : 'true'
+}
+
+Set.prototype.remove = function(indexNum){
+  console.log(this.dataStore[indexNum]+' is removed.')
+  this.dataStore.splice(indexNum,1)
+  this.size--
+}
+
+Set.prototype.show = function(indexNum){
+  return this.dataStore[indexNum]
+}
+
+Set.prototype.length = function(){
+  return this.size
+}
+
+Set.prototype.showAll = function(){
+  const elementsInStore = []
+  for(i=0; i < this.size; i++){
+    elementsInStore.push(this.dataStore[i])
+  }
+  return elementsInStore
+}
+
+Set.prototype.forEach = function(func){
+  const elementsInStore = []
+  for(i=0; i < this.size; i++){
+    elementsInStore.push(func(this.dataStore[i]))
+  }
+  return elementsInStore
+}
+
+Set.prototype.union = function(secondSet){
+  const union = []
+   union.push(this.dataStore)
+   union.push(secondSet.dataStore)
+  return union
+}
+
+
+
+
+Set.prototype.intersect = function(secondSet){
+
+  const intersectorsStore = []
+  if(secondSet.dataStore > this.dataStore){
+    for(i=0; i <= secondSet.size; i++){
+      for(x=0; x < this.size; x++){
+        if(this.dataStore[i] == secondSet.dataStore[x]){
+          intersectorsStore.push(this.dataStore[i])
+        }
+      }
+    }
+    return intersectorsStore
+
+  } else {
+
+    for(i=0; i <= this.size; i++){
+      for(x=0; x < secondSet.size; x++){
+        if(secondSet.dataStore[i] == this.dataStore[x]){
+          intersectorsStore.push(secondSet.dataStore[i])
+        }
+      }
+    }
+    return intersectorsStore
+  }
+
+}
+
+/* ~~~~~~~~~~~~~~~~~ THE holy shrine of checkFrequency ~~~~~~~~~~~~~~~~~~~~~~~~ */
+
+Set.prototype.checkFrequency = function(frequency){
+  const setArr = this.dataStore
+  const solution = []
+  let freqOfAppearance = 0
+
+  for(i=0; i<=setArr.length; i++){
+    freqOfAppearance = 0
+    for(x=i; x < setArr.length; x++){
+      if(setArr[i] == setArr[x]){
+        freqOfAppearance++
+      } else {
+        //i = i+1
+        i = x
+        x = i-1
+        freqOfAppearance = 0
+      }
+      if(freqOfAppearance > frequency-1){
```

Comment:
This is a touch unreadable for me, so maybe I'm exposing some ignorance, what is this doing?